utils/config-generator.py

```diff
@@ -52,8 +52,11 @@
 )
 
 # If there are no genesis params, this is a public chain.
-THIS_IS_A_PUBLIC_NET = True if not NETWORK_CONFIG.get("genesis") else False
-
+THIS_IS_A_PUBLIC_NET = True if ((not NETWORK_CONFIG.get("genesis"))) else False
+try:
+    JOIN_PUBLIC_NETWORK =  CHAIN_PARAMS.get("join_public_network")
+except:
+    JOIN_PUBLIC_NETWORK =  THIS_IS_A_PUBLIC_NET
```

Comment:
`.get` shouldn't ever throw an error. I don't think the try/except is necessary and it should be able to be simplified to:
```suggestion
JOIN_PUBLIC_NETWORK =  CHAIN_PARAMS.get("join_public_network", THIS_IS_A_PUBLIC_NET)
```

Reply:
I think we also need to do a more sophisticated check. Say `THIS_IS_A_PUBLIC_NET == False`, so we are spawning a new chain. But the user also set `JOIN_PUBLIC_NETWORK == True`. So it is a contradiction. We should probably throw an exception in this case.

Reply:
Indeed. I amended my proposition.

Reply:
I'm not familiar with helm, so I don't know if we can push the check in the template, in order to fail earlier in the process ?